src/main/kotlin/com/openlattice/launchpad/configuration/IntegrationConfiguration.kt

```diff
@@ -108,9 +124,120 @@ data class Integration(
         @JsonProperty(SOURCE) val source: String,
         @JsonProperty(DESTINATION) val destination: String,
         @JsonProperty(MERGE_SQL) val mergeSql: String = "",
-        @JsonProperty(MASTER_TABLE_SQL) val masterTableSql : String = "",
+        @JsonProperty(MASTER_TABLE_SQL) val masterSql : String = "",
         @JsonProperty(GLUTTONY) val gluttony : Boolean = false
-)
+): Transferable {
+    @JsonIgnore
+    override fun getLaunchPad(): String {
+        return source
+    }
+
+    @JsonIgnore
+    override fun getLandingPad(): String {
+        return destination
+    }
+
+    @JsonIgnore
+    override fun getMasterTableSql(): String {
+        return masterSql
+    }
+
+    @JsonIgnore
+    override fun getMergeTableSql(): String {
+        return mergeSql
+    }
+
+    @JsonIgnore
+    override fun getSourceName(): String {
+        return source
+    }
+
+    @JsonIgnore
+    override fun getBucketColumn(): String? {
+        return null
+    }
+}
+
+interface Transferable {
+    @JsonIgnore
+    fun getMasterTableSql(): String
+
+    @JsonIgnore
+    fun getMergeTableSql(): String
+
+    @JsonIgnore
+    fun getSourceName(): String
+
+    @JsonIgnore
+    fun getLandingPad(): String
+
+    @JsonIgnore
+    fun getLaunchPad(): String
+
+    @JsonIgnore
+    fun getBucketColumn(): String?
+}
+
+const val ARCHIVES = "archives"
+const val STRATEGY = "strategy"
+const val CONSTRAINTS = "constraints"
+const val BUCKET_COLUMN = "column"
+const val INTERVAL_COUNT = "intervalCount"
+const val INTERVAL_UNIT = "intervalUnit"
+
+data class Archive(
+        @JsonProperty(SOURCE) val source: String,
+        @JsonProperty(STRATEGY) val strategy: DailyBucketingArchiveStrategy,
+        @JsonProperty(DESCRIPTION) val description : String = "",
+        @JsonProperty(DESTINATION) val destination: String
+): Transferable {
+    @JsonIgnore
+    override fun getLaunchPad(): String {
+        return getArchiveSql()
+    }
+
+    @JsonIgnore
+    override fun getMasterTableSql(): String {
+        return ""
+    }
+
+    @JsonIgnore
+    override fun getMergeTableSql(): String {
+        return ""
```

Comment:
🧐

Reply:
It's not used in the archive flow either